boot/freeldr/bootsect/btrfs.S

```diff
@@ -0,0 +1,790 @@
+/*
+ * COPYRIGHT:       See COPYING in the top level directory
+ * PROJECT:         ReactOS Bootsector
+ * FILE:            boot/freeldr/bootsect/btrfs.S
+ * PURPOSE:         BTRFS boot sector
+ * PROGRAMMERS:     Victor Perevertkin
+ */
+
+#include <asm.inc>
+#include <freeldr/include/arch/pc/x86common.h>
+
+SizeOfDataArea        = 32
+
+/* Put the stack below the data area */
+BootSectorStackTop    = (HEX(7c00) - SizeOfDataArea)
+
+/* Data area offsets for uninitialized data */
+DataAreaStart         = BootSectorStackTop + 0  /* dword */
+BiosCHSDriveSize      = BootSectorStackTop + 8  /* dword */
+LBASectorsRead        = BootSectorStackTop + 12 /* word */
+ReadSectorsOffset     = BootSectorStackTop + 16 /* word */
+ReadClusterOffset     = BootSectorStackTop + 18 /* word */
+PutCharsOffset        = BootSectorStackTop + 20 /* word */
+//BootDrive             = BootSectorStackTop + 22 /* byte */
+//PartitionStartSector  = BootSectorStackTop + 23 /* byte */
+
+/* Macro for bp relative memory access to reduce code size */
+#define BP_REL(x) ss:[bp + x - BootSectorStackTop]
+
+.code16
+
+CHUNK_MAP_OFFSET = HEX(8200) // max - 256 chunks (chunk is 24 bytes)
+DATA_BUFFER_SEGMENT = HEX(9a0) // here we will store FS structures read from disk
+
+FIRST_SUPERBLOCK_OFFSET = HEX(80) //in sectors
+
+ROOT_TREE_OBJECTID = 1
+EXTENT_TREE_OBJECTID = 2
+CHUNK_TREE_OBJECTID = 3
+DEV_TREE_OBJECTID = 4
+FS_TREE_OBJECTID = 5
+FIRST_CHUNK_TREE_OBJECTID = 256
+
+DIR_ITEM_KEY = 84
+EXTENT_DATA_KEY = 108
+ROOT_ITEM_KEY = 132
+EXTENT_ITEM_KEY = 168
+CHUNK_ITEM_KEY = 228
+
+BTRFS_FT_REG_FILE = 1
+
+KEY_SIZE = 17
+MAX_CHUNK_ITEMS = 255
+
+start:
+    jmp short main
+    nop
+// globals
+ChunkMapSize:
+    .byte 0
+BootDrive:
+    .byte 0
+PartitionStartLBA:
+    .quad HEX(3f) // hardcoded! usetup has to write proper value here during installation
```

Comment:
So it's the default "63" sectors (you may add this in the comment).

Reply:
Right now it's being always overwritten in installer